File: Scripts/results/generated_code_results/ablation_study/sam/Productivity/TaskManagerApp-UI/BoardScreen.js
import React from 'react';

import { View, Text, StyleSheet, TouchableOpacity, Image, ScrollView } from 'react-native';

import Icon from 'react-native-vector-icons/MaterialIcons';

import { useNavigation } from '@react-navigation/native';



const BoardScreen = () => {

  const navigation = useNavigation();



  const boards = [

    { name: 'Personal', icon: 'person', tasks: 17 },

    { name: 'Work', icon: 'work', tasks: 10 },

    { name: 'Private', icon: 'lock', tasks: 2 },

    { name: 'Meeting', icon: 'people', tasks: 5 },

    { name: 'Events', icon: 'event', tasks: 3 },

    { name: 'Create Board', icon: 'add', tasks: null },

  ];



  const renderBoard = (board, index) => (

    <TouchableOpacity key={index} style={styles.boardItem}>

      <View style={styles.iconContainer}>

        <Icon name={board.icon} size={24} color="#6200EE" />

      </View>

      <Text style={styles.boardName}>{board.name}</Text>

      {board.tasks !== null && (

        <Text style={styles.taskCount}>{board.tasks} Task</Text>

      )}

    </TouchableOpacity>

  );



  return (

    <View style={styles.container}>

      <View style={styles.header}>

        <Text style={styles.title}>Board</Text>

        <Image

          source={require('../assets/snack-icon.png')}

          style={styles.avatar}

        />

      </View>

      <ScrollView contentContainerStyle={styles.boardsContainer}>

        {boards.map(renderBoard)}

      </ScrollView>

      <View style={styles.bottomNav}>

        <TouchableOpacity onPress={() => navigation.navigate('Board')}>

          <Icon name="grid-view" size={24} color="#6200EE" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Dashboard')}>

          <Icon name="check" size={24} color="#757575" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('AddTask')} style={styles.addButton}>

          <Icon name="add" size={24} color="#FFFFFF" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Task')}>

          <Icon name="description" size={24} color="#757575" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Message')}>

          <Icon name="chat" size={24} color="#757575" />

        </TouchableOpacity>

      </View>

    </View>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#F5F5F5',

  },

  header: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    padding: 16,

    backgroundColor: '#FFFFFF',

  },

  title: {

    fontSize: 24,

    fontWeight: 'bold',

  },

  avatar: {

    width: 40,

    height: 40,

    borderRadius: 20,

  },

  boardsContainer: {

    flexDirection: 'row',

    flexWrap: 'wrap',

    justifyContent: 'space-between',

    padding: 16,

  },

  boardItem: {

    width: '48%',

    backgroundColor: '#FFFFFF',

    borderRadius: 8,

    padding: 16,

    marginBottom: 16,

    alignItems: 'center',

  },

  iconContainer: {

    width: 48,

    height: 48,

    borderRadius: 24,

    backgroundColor: '#E8E0FF',

    justifyContent: 'center',

    alignItems: 'center',

    marginBottom: 8,

  },

  boardName: {

    fontSize: 16,

    fontWeight: 'bold',

    marginBottom: 4,

  },

  taskCount: {

    fontSize: 14,

    color: '#757575',

  },

  bottomNav: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    alignItems: 'center',

    backgroundColor: '#FFFFFF',

    paddingVertical: 12,

    borderTopWidth: 1,

    borderTopColor: '#E0E0E0',

  },

  addButton: {

    width: 56,

    height: 56,

    borderRadius: 28,

    backgroundColor: '#6200EE',

    justifyContent: 'center',

    alignItems: 'center',

    marginTop: -28,

  },

});



export default BoardScreen;

File: Scripts/results/generated_code_results/Flutter/Productivity/TaskManagerApp-UI/BoardScreen.dart
import 'package:flutter/material.dart';
class BoardScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Board', style: TextStyle(color: Colors.black, fontWeight: FontWeight.bold)),
backgroundColor: Colors.transparent,
elevation: 0,
actions: [
Padding(
padding: const EdgeInsets.all(8.0),
child: CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
),
],
),
body: GridView.count(
crossAxisCount: 2,
padding: EdgeInsets.all(16),
childAspectRatio: 1.2,
children: [
_buildBoardCard(context, 'Personal', Icons.person, '17 Task', Colors.blue),
_buildBoardCard(context, 'Work', Icons.work, '10 Task', Colors.orange),
_buildBoardCard(context, 'Private', Icons.lock, '2 Task', Colors.green),
_buildBoardCard(context, 'Meeting', Icons.people, '5 Task', Colors.purple),
_buildBoardCard(context, 'Events', Icons.event, '3 Task', Colors.red),
_buildCreateBoardCard(context),
],
),
bottomNavigationBar: BottomAppBar(
shape: CircularNotchedRectangle(),
notchMargin: 8.0,
child: Row(
mainAxisAlignment: MainAxisAlignment.spaceAround,
children: [
IconButton(icon: Icon(Icons.grid_view), onPressed: () {}),
IconButton(
icon: Icon(Icons.check_circle_outline),
onPressed: () => Navigator.pushNamed(context, 'Dashboard'),
),
SizedBox(width: 48),
IconButton(
icon: Icon(Icons.description_outlined),
onPressed: () => Navigator.pushNamed(context, 'Task'),
),
IconButton(
icon: Icon(Icons.chat_bubble_outline),
onPressed: () => Navigator.pushNamed(context, 'Message'),
),
],
),
),
floatingActionButton: FloatingActionButton(
child: Icon(Icons.add),
backgroundColor: Colors.orange,
onPressed: () => Navigator.pushNamed(context, 'AddTask'),
),
floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
);
}
Widget _buildBoardCard(BuildContext context, String title, IconData icon, String taskCount, Color color) {
return Card(
elevation: 2,
shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
child: InkWell(
onTap: () {
// TODO: Implement board specific action
},
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
mainAxisAlignment: MainAxisAlignment.center,
children: [
Container(
padding: EdgeInsets.all(12),
decoration: BoxDecoration(
color: color.withOpacity(0.1),
borderRadius: BorderRadius.circular(12),
),
child: Icon(icon, color: color, size: 32),
),
SizedBox(height: 16),
Text(title, style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
SizedBox(height: 8),
Text(taskCount, style: TextStyle(color: Colors.grey)),
],
),
),
),
);
}
Widget _buildCreateBoardCard(BuildContext context) {
return Card(
elevation: 2,
shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
child: InkWell(
onTap: () {
// TODO: Implement create board action
},
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
mainAxisAlignment: MainAxisAlignment.center,
children: [
Container(
padding: EdgeInsets.all(12),
decoration: BoxDecoration(
color: Colors.green.withOpacity(0.1),
borderRadius: BorderRadius.circular(12),
),
child: Icon(Icons.add, color: Colors.green, size: 32),
),
SizedBox(height: 16),
Text('Create Board', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
],
),
),
),
);
}
}